phylo/src/optimisers/branch_length_optimiser.rs

```diff
@@ -0,0 +1,29 @@
+use argmin::core::{CostFunction, Executor, IterState, State};
+use argmin::solver::brent::BrentOpt;
+use log::{debug, info, warn};
+
+use crate::substitution_models::{SubstitutionLikelihoodCost, SubstitutionModel};
+use crate::tree::NodeIdx;
+use crate::Result;
+
+pub(crate) struct SingleBranchOptimiser<'a> {
+    pub(crate) likelihood_cost: &'a SubstitutionLikelihoodCost<'a, 4>,
```

Comment:
nit: Since you use constants, explaining them with comments/doc-strings makes sense

Reply:
Agreed. This will be generalised away in the next PR, but I will explain constants where they still appear.